File: lista de exercicios/questão 4/questao.py
class Animal:
    def som(self):
        print("Som de animal")

class Cachorro(Animal):
    def som(self):
        print("O cachorro faz: Au Au")

class Gato(Animal):
    def som(self):
        print("O gato faz: Miau")

cachorro = Cachorro()
gato = Gato()

cachorro.som()
gato.som()

File: lista de exercicios/questão 4/questao.go
package main

import "fmt"

type Animal interface {
    Som()
}

type Cachorro struct{}

func (Cachorro) Som() {
    fmt.Println("O cachorro faz: Au Au")
}

type Gato struct{}

func (Gato) Som() {
    fmt.Println("O gato faz: Miau")
}

func main() {
    var animal Animal

    animal = Cachorro{}
    animal.Som()

    animal = Gato{}
    animal.Som()
}
